scripts/quests/otherAreas/Rycharde_the_Chef.lua

```diff
@@ -0,0 +1,146 @@
+                end,
+            },
+        },
+    },
+
+    -- Section: Get quest.
+    {
+        check = function(player, status, vars)
+            return status == QUEST_AVAILABLE and vars.Prog > 1
+        end,
+
+        [xi.zone.MHAURA] =
+        {
+            ['Rycharde'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 2 then
+                        return quest:progressEvent(70, xi.items.DHAMEL_MEAT)
+                    elseif quest:getVar(player, 'Prog') == 3 then -- You rejected the quest and you ask again.
+                        return quest:progressEvent(71, xi.items.DHAMEL_MEAT)
+                    end
+                end,
+            },
+
+            ['Take'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 2 then
+                        return quest:event(68)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [70] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 3)
+                    if (option == 71 or option == 72) then    --70 = answer no; 71 = answer yes!
+                        quest:begin(player)
+                    end
+                end,
+                [71] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 3)
+                    if (option == 71 or option == 72) then    --70 = answer no; 71 = answer yes!
+                        quest:begin(player)
+                    end
+                end,
+            },
+        },
+    },
+
+    -- Section: Actual questing.
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.MHAURA] =
+        {
+            ['Rycharde'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 3 then
+                        return quest:event(72)
+                    end
+                end,
+
+                onTrade = function(player, npc, trade)
+                    local tradeCount  = trade:getItemCount()
+                    local dhalmelMeat = trade:hasItemQty(xi.items.DHAMEL_MEAT, trade:getItemCount())
+                    if dhalmelMeat  == true and tradeCount == 2 then
+                        return quest:progressEvent(74) -- Completed OK
+                    elseif dhalmelMeat  == true and tradeCount == 1 then
+                        return quest:event(73) -- That's not enough!
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [74] = function(player, csid, option, npc)
+                    player:confirmTrade()
+                    quest:complete(player)
+                    quest:setVar(player, 'Prog', 0)
```

Comment:
Is the quest:setVar handled by cleanup from quest:complete()?  Also, are those other charvars used?

Reply:
the other 2 vars are there to start the next quest in line and handled there
about the quest:complete, i honestly dont know, i did that out of habit